### PR TITLE
[SDK-1169] Deprecate AppboyLocationService.requestInitialization(context)

### PR DESCRIPTION
**Pull request by chshapiro**
https://jira.braze.com/browse/SDK-1196

### CODE REVIEW

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 1]
```diff
@@ -84,14 +84,14 @@ Add the following call to notify Braze when a user grants the location permissio
 {% tab JAVA %}
```
Comment (Bucimis) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:84]: I think we should add an alert (see e.g. line 67) saying something like:

"If you are using Braze Android SDK version PLACEHOLDER or below, use `AppboyLocationService.requestInitialization(context)` instead"

And we can update the placeholder at release time before merge

**Review by Bucimis — APPROVED** [review]

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 2]
```diff
@@ -84,14 +84,14 @@ Add the following call to notify Braze when a user grants the location permissio
...
 
 ```java
-AppboyLocationService.requestInitialization(context);
+Appboy.getInstance(context).requestLocationInitialization();
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:87]:
```suggestion
Braze.getInstance(context).requestLocationInitialization();
```

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 3]
```diff
@@ -84,14 +84,14 @@ Add the following call to notify Braze when a user grants the location permissio
...
 ```
 
 {% endtab %}
 {% tab KOTLIN %}
 
 ```kotlin
-AppboyLocationService.requestInitialization(context)
+Appboy.getInstance(context).requestLocationInitialization()
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:94]:
```suggestion
Braze.getInstance(context).requestLocationInitialization()
```

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 4]
```diff
@@ -117,7 +117,7 @@ public class RuntimePermissionUtils {
         if (areAllPermissionsGranted(grantResults)) {
           Log.i(TAG, "Required location permissions granted.");
           Toast.makeText(context, "Required location permissions granted.", Toast.LENGTH_SHORT).show();
-          AppboyLocationService.requestInitialization(context);
+          Appboy.getInstance(context).requestLocationInitialization();
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:120]:
```suggestion
          Braze.getInstance(context).requestLocationInitialization();
```

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md [hunk 5]
```diff
@@ -155,7 +155,7 @@ object RuntimePermissionUtils {
         if (areAllPermissionsGranted(grantResults)) {
           Log.i(TAG, "Required location permissions granted.")
           Toast.makeText(context, "Required location permissions granted.", Toast.LENGTH_SHORT).show()
-          AppboyLocationService.requestInitialization(context)
+          Appboy.getInstance(context).requestLocationInitialization()
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:158]:
```suggestion
          Braze.getInstance(context).requestLocationInitialization()
```

File: _docs/_developer_guide/platform_integration_guides/fireos/advanced_use_cases/locations_and_geofences.md [hunk 6]
```diff
@@ -78,7 +78,7 @@ For Android M and higher versions, you must request location permissions from th
 Add the following call to notify Braze when a user grants the location permission to your app:
 
 ```
-AppboyLocationService.requestInitialization(context);
+Appboy.getInstance(context).requestLocationInitialization();
```
Comment (radixdev) [inline _docs/_developer_guide/platform_integration_guides/fireos/advanced_use_cases/locations_and_geofences.md:81]:
```suggestion
Braze.getInstance(context).requestLocationInitialization();
```

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md (hunk 1 above)
Comment (radixdev) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:84]: I think we can make a reasonable assumption that a client reading these docs and integrating will be on the latest SDK version and thus won't need this warning. Adding references to deprecated code to our documentation seems messy to me, but up to you.

**Review by radixdev — DISMISSED** [review]

File: _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md (hunk 1 above)
Comment (Bucimis) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:84]: yeah there's a balance there, I don't mind experimenting this time with not adding the old method

Comment (chshapiro) [reply _docs/_developer_guide/platform_integration_guides/android/advanced_use_cases/locations_and_geofences.md:84]: 👍 I think recently we've done that with iOS as well for some smaller things

**Comment by KellieHawks** [comment]
@chshapiro Is this PR good to merge?

**Comment by chshapiro** [comment]
yes thank you!

**Review by KellieHawks — APPROVED** [review]
lgtm!

**Pull request merged**